### PR TITLE
ci: doc to gh actions

### PR DESCRIPTION
**Pull request by holmanb**
```
test: move doc test from travis to actions

Changes:
- drop unnecessary setuptools pin
- use current RTD Python version 3.10.8
```

## Additional Context
Makes doc failures visible sooner via gh actions, plus reduce CI bill. This is mostly just copy / paste from existing gh actions tests and the current travis test - low impact / low effort.

FYI @s-makin I mentioned this would be possible today while we were wrapping up #1933.

### CODE REVIEW

File: .github/workflows/check_format.yml [hunk 1]
```diff
@@ -62,3 +62,26 @@ jobs:
...
+          git fetch --unshallow
+          git remote add upstream https://git.launchpad.net/cloud-init
+
+      - name: "1. Install dependencies 2. Spellcheck 3. Build docs"
+        env:
+          TOXENV: doc
+        run: |
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -qy update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -qy install tox lintian
+          make check_spelling
```
Comment (TheRealFalcon) [inline .github/workflows/check_format.yml:86]: Nit: these last two lines would be better as their own run commands.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
This might be a little pedantic, but I think we should ensure we're using the same python version that rtd uses to build the docs. In the past, because of dependency issues, we've had issues where local/CI passes but rtd fails on a different python version. That's probably much less likely to happen after removing the old sphinx pins, but I still think it'd make our CI better.

**Review by s-makin — APPROVED** [review]
Definitely a nice idea, avoiding the "install python" step that it does every time. Anything that makes the process faster or more efficient gets a thumbs up from me.

**Comment by holmanb** [comment]
> This might be a little pedantic, but I think we should ensure we're using the same python version that rtd uses to build the docs. In the past, because of dependency issues, we've had issues where local/CI passes but rtd fails on a different python version. That's probably much less likely to happen after removing the old sphinx pins, but I still think it'd make our CI better.

I will do that, but it may drift since RTD only allows configuring [Major.Minor version](https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools-python). I see the most recent [build used 3.10.8](https://readthedocs.org/projects/cloudinit/builds/19129720/).

**Review by TheRealFalcon — APPROVED** [review]
LGTM. With the python version I'm mostly concerned about major.minor.

**Comment by holmanb** [comment]
> LGTM. With the python version I'm mostly concerned about major.minor

Ah, I guess I was confused about that then, since Jammy already had the same major.minor as RTD.

**Pull request merged**